rasterio/_base.pyx

```diff
@@ -763,6 +764,20 @@ cdef class DatasetBase(object):
                 stacklevel=2)
             return Affine.from_gdal(*self.get_transform())
 
+    def subdatasets(self):
```

Comment:
Per our discussion in https://github.com/mapbox/rasterio/pull/1134, this should be a property.